plugins/modules/aci_l3out.py

```diff
@@ -307,26 +386,84 @@ def main():
     enforceRtctrl = module.params.get("route_control")
     vrf = module.params.get("vrf")
     l3protocol = module.params.get("l3protocol")
+    ospf = module.params.get("ospf")
     asn = module.params.get("asn")
     state = module.params.get("state")
     tenant = module.params.get("tenant")
     name_alias = module.params.get("name_alias")
 
     if l3protocol:
         if "eigrp" in l3protocol and asn is None:
-            module.fail_json(msg="Parameter 'asn' is required when l3protocol is 'eigrp'")
+            module.fail_json(
+                msg="Parameter 'asn' is required when l3protocol is 'eigrp'"
+            )
         if "eigrp" not in l3protocol and asn is not None:
-            module.warn("Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored")
+            module.warn(
+                "Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored"
+            )
 
     enforce_ctrl = ""
     if enforceRtctrl is not None:
         if len(enforceRtctrl) == 1 and enforceRtctrl[0] == "import":
-            aci.fail_json("The route_control parameter is invalid: allowed options are export or import,export only")
+            aci.fail_json(
+                "The route_control parameter is invalid: allowed options are export or import,export only"
+            )
         elif len(enforceRtctrl) == 1 and enforceRtctrl[0] == "export":
             enforce_ctrl = "export"
         else:
             enforce_ctrl = "export,import"
-    child_classes = ["l3extRsL3DomAtt", "l3extRsEctx", "bgpExtP", "ospfExtP", "eigrpExtP", "pimExtP"]
+    else:
+        enforce_ctrl = "export"
```

Comment:
Is this else required? since the default when None is provided to apic is export. Would likely require some change in then var assignments.

Reply:
it is not required as you clearly explained but I just realized that the previous author added this variable `enforce_ctrl = ""`
 This why I added this else statement but will change it.